qadence2_expressions/core/utils.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+from typing import Any, Protocol, runtime_checkable
+
+
+@runtime_checkable
+class Numeric(Protocol):
+    """
+    Protocol class to represent Python base classes for numeric values,
+    such as complex, float, int, but also include packages types, such
+    as numpy and torch. It is especially useful for handling operations
+    with Expressions and any kind of "Numeric" type.
```

Comment:
Since the expressions only handle Python inputs, what is the point of adding support to `numpy` or `torch` at this point?

Reply:
to be backwards compatible